### PR TITLE
Move setup_envars recipe to aws-parallelcluster-common cookbook

### PR DESCRIPTION
**Pull request by awslitvit**
### Description of changes
* setup_envars recipe moved to aws-parallelcluster-common cookbook
* Isolated test for recipes created in aws-parallelcluster-common cookbook
* recipe tests for aws-parallelcluster-common cookbook added to github actions

### Tests
Tested on Docker and EC2

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [x] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [ ] Check if documentation is impacted by this change.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -85,6 +85,32 @@ jobs:
...
+        uses: actionshub/test-kitchen@main
+        with:
+          os: ${{ matrix.os }}
+        env:
+          CHEF_LICENSE: accept-no-persist
+          KITCHEN_YAML: kitchen.docker.yml
+          KITCHEN_LOCAL_YAML: cookbooks/aws-parallelcluster-common/kitchen.recipes.yml
+          KITCHEN_GLOBAL_YAML: kitchen.global.yml
+        continue-on-error: false
   dokken-recipes:
```
Comment (eantonin) [inline .github/workflows/ci.yml:114]: nitpick, should we rename as dokken-install-recipes?

File: cookbooks/aws-parallelcluster-common/test/recipes/controls/setup_envars_spec.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+control 'path_contains_required_directories' do
+  title 'System path contains required directories'
+
+  describe file('/etc/profile.d/path.sh') do
+    it { should exist }
+    its('mode') { should cmp '0755' }
+    its('owner') { should eq 'root' }
+    its('group') { should eq 'root' }
+    its('content') { should match 'PATH_REQUIRED_DIRECTORIES=\(/usr/local/sbin /usr/local/bin /sbin /bin /usr/sbin /usr/bin /opt/aws/bin\)' }
```
Comment (eantonin) [inline cookbooks/aws-parallelcluster-common/test/recipes/controls/setup_envars_spec.rb]: question: here we check the path in the file, but I guess we should check also the env PATH variable, am I wrong?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (awslitvit) [reply .github/workflows/ci.yml:114]: It's all still temporary, I would not invest time in fixing something we know we are going to change soon.

File: cookbooks/aws-parallelcluster-common/test/recipes/controls/setup_envars_spec.rb (hunk 2 above)
Comment (awslitvit) [reply cookbooks/aws-parallelcluster-common/test/recipes/controls/setup_envars_spec.rb]: Unfortunately, the variable is not set at this point.
However, I already have another change with a workaround to test it, I can actually add it to this PR..

**Review by eantonin — APPROVED** [review]
LGTM

**Pull request merged**